### PR TITLE
[MODCLUSTER-584] Enclose IPv6 address into square brackets

### PR DESCRIPTION
**Pull request by mmadzin**
Enclose IPv6 address into square brackets while sending STATUS info to balancer.

MODCLUSTER-584: https://issues.jboss.org/browse/MODCLUSTER-584

@rhusar - seems that modcluster is trying to resolve IP address to hostname (seems it's another suspect for the automagic plague)

### CODE REVIEW

**Comment by Karm** [comment]
@mmadzin Please, reference the JIRA number in the commit message: [MODCLUSTER-584](https://issues.jboss.org/browse/MODCLUSTER-584) and preferably also in the summary of the PR.

**Comment by rhusar** [comment]
@mmadzin Correct, please always structure commit message and github summary for that matter by "MODCLUSTER-<number> <Issue description>".

**Comment by Karm** [comment]
@mmadzin Amend the commit message too, please. It still says ```Enclose IPv6 address into square brackets```. Changing the PR subject is not enough 🙁

**Comment by rhusar** [comment]
Just in case... ;-)

`git commit --amend`
`git push origin --force`

**Comment by mmadzin** [comment]
@paul Commit message was fixed.

File: core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java [hunk 1]
```diff
@@ -600,7 +600,16 @@ private String sendRequest(MCMPRequest request, Proxy proxy) {
         synchronized (proxy) {
             try {
                 String line = null;
-                String proxyhead = head + proxy.getSocketAddress().getHostName() + ":" + proxy.getSocketAddress().getPort();
+                String hostAddress = proxy.getSocketAddress().getAddress().getHostAddress();
+                String hostName = proxy.getSocketAddress().getHostName();
```
Comment (pferraro) [inline core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java]: InetSocketAddress.getHostName() will perform a reverse DNS lookup.  See:
http://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#getHostName--
We should instead use http://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#getHostString--

File: core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java [hunk 2]
```diff
@@ -600,7 +600,16 @@ private String sendRequest(MCMPRequest request, Proxy proxy) {
         synchronized (proxy) {
             try {
                 String line = null;
-                String proxyhead = head + proxy.getSocketAddress().getHostName() + ":" + proxy.getSocketAddress().getPort();
+                String hostAddress = proxy.getSocketAddress().getAddress().getHostAddress();
```
Comment (rhusar) [inline core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java]: I wonder if this is needed, why not just?

```java
                String hostString = proxy.getSocketAddress().getHostString();
                String proxyhead;

                if (hostString.contains(":")) {
                    proxyhead = head + "[" + hostString + "]:" + proxy.getSocketAddress().getPort();
                } else {
                    proxyhead = head + hostString + ":" + proxy.getSocketAddress().getPort();
                }
```

**Comment by mmadzin** [comment]
@rhusar , @pferraro 
For master, there are two commits, the first for removing the [MODCLUSTER-584] issue and the second to change the behavior with DNS lookup.

**Pull request merged**